### PR TITLE
Fix for mappings with holes within erase sector range

### LINKED ISSUE #39: picotool crashes when given an ELF built with embedded rust
```console
$ cargo build --example pico_blinky
   Compiling pico v0.1.0 (/home/jonathan/rp-hal/boards/pico)
    Finished dev [unoptimized + debuginfo] target(s) in 0.91s
$ ../picotool/build/picotool info ./target/thumbv6m-none-eabi/debug/examples/pico_blinky
ERROR: filename './target/thumbv6m-none-eabi/debug/examples/pico_blinky' does not have a recognized file type (extension)
$ cp ./target/thumbv6m-none-eabi/debug/examples/pico_blinky ./pico_blinky.elf
$ ../picotool/build/picotool info ./pico_blinky.elf
picotool: /home/jonathan/picotool/main.cpp:156: void range_map<T>::check_overlap(uint32_t) [with T = long unsigned int; uint32_t = unsigned int]: Assertion `p >= f->first' failed.
[1]    3871 abort      ../picotool/build/picotool info ./pico_blinky.elf
```

### PR DESCRIPTION
**Pull request by jannic**
Fixes #39

### CODE REVIEW

**Comment by jannic** [comment]
Please review carefully: My C++ is a little bit rusty :-)

File: main.cpp [hunk 1]
```diff
@@ -854,14 +854,22 @@ struct file_memory_access : public memory_access {
...
-            assert( this_size);
-            fseek(file, result.second + result.first.offset, SEEK_SET);
-            fread(buffer, this_size, 1, file);
-            buffer += this_size;
-            address += this_size;
-            size -= this_size;
+            try {
+                auto result = rmap.get(address);
+                uint this_size = std::min(size, result.first.max_offset - result.first.offset);
+                printf("this_size is 0x%08x\n", this_size);
```
Comment (lurch) [inline main.cpp:860]: Is this a leftover debugging-print?

Comment (jannic) [reply main.cpp]: Ok, removed the printf statement and made the gap-spanning coalescing conditional on the memory type.

**Comment by kilograham** [comment]
Thanks; i have closed this in favor of #68

**Comment by jannic** [comment]
Ok, I just verified that #68 fixes it for me as well. :+1:

**Pull request closed without merging**